examples/golang/topic-creation/topic-creation.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/sns"
+)
+
+// SNSCreateTopicAPI defines the interface for the CreateTopic function.
+// We use this interface to test the function using a mocked service.
+type SNSCreateTopicAPI interface {
+	CreateTopic(ctx context.Context,
+		params *sns.CreateTopicInput,
+		optFns ...func(*sns.Options)) (*sns.CreateTopicOutput, error)
+}
+
+// MakeTopic creates an Amazon Simple Notification Service (Amazon SNS) topic.
+// Inputs:
+//     c is the context of the method call, which includes the AWS Region.
+//     api is the interface that defines the method call.
+//     input defines the input arguments to the service call.
+// Output:
+//     If success, a CreateTopicOutput object containing the result of the service call and nil.
+//     Otherwise, nil and an error from the call to CreateTopic.
+func MakeTopic(c context.Context, api SNSCreateTopicAPI, input *sns.CreateTopicInput) (*sns.CreateTopicOutput, error) {
+	return api.CreateTopic(c, input)
+}
+
+func main() {
+	topic := flag.String("t", "", "The name of the topic")
+	attributes := flag.String("a", "", "Topic attributes needed")
+	attributesmap := map[string]string{}
+	json.Unmarshal([]byte(*attributes), &attributesmap) // convert JSON string to Go map
+	flag.Parse()
+
+	if *topic == "" {
+		fmt.Println("You must supply the name of the topic")
+		fmt.Println("-t TOPIC")
+		return
+	}
+
+	access_key := "0555b35654ad1656d804"
+	secret_key := "h7GhxuBLTrlhVUyxSPUKUV8r/2EI4ngqJxD7iBdBYLhwluN30JaT3Q=="
+	token_id := ""
+	url := "http://127.0.0.1:8000"
+
+	customResolver := aws.EndpointResolverWithOptionsFunc(func(service, region string, options ...interface{}) (aws.Endpoint, error) {
+		if service == sns.ServiceID {
+			return aws.Endpoint{
+				URL: url,
+			}, nil
+		}
+		return aws.Endpoint{}, fmt.Errorf("unknown endpoint requested")
+	})
+
+	cfg, err := config.LoadDefaultConfig(context.TODO(),
+		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(access_key, secret_key, token_id)),
+		config.WithEndpointResolverWithOptions(customResolver),
+	)
+	if err != nil {
+		panic("configuration error, " + err.Error())
+	}
+
+	client := sns.NewFromConfig(cfg)
+
+	input := &sns.CreateTopicInput{
+		Name:       topic,
+		Attributes: attributesmap,
+	}
+
+	results, err := MakeTopic(context.TODO(), client, input)
+	if err != nil {
+		fmt.Println(err.Error())
```

Comment:
can you please unify the error handling and printing between `object-upload.go` and `topic-creation.go`
as well as the printing

Reply:
Will do. Also, does the SDK version matter?
Should I use sdk v1 or sdk v2 on the code samples?

Reply:
lets use v1 in the examples